### PR TITLE
Allow OPTIONS request for issue #22

### PR DESCRIPTION
**Pull request by ansariabr**
Allow OPTIONS request for issue [#22](https://github.com/natemcmaster/dotnet-serve/issues/22)

### CODE REVIEW

File: src/dotnet-serve/Startup.cs [hunk 1]
```diff
@@ -62,6 +63,12 @@ public void ConfigureServices(IServiceCollection services)
 
         public void Configure(IApplicationBuilder app)
         {
+            app.UseCors(corsPolicy =>
```
Comment (natemcmaster) [inline src/dotnet-serve/Startup.cs]: This makes the server much more permissive by default. Maybe this should be opt-in. What do you think about `dotnet serve --cors`?

Comment (ansariabr) [reply src/dotnet-serve/Startup.cs]: I quickly checked nginx's default configuration, and it does not allow OPTIONS out of the box. It makes sense to make it opt-in by using `--cors` option.
And I would suggest that this option will by default allow all origin and methods. For configuring selected allowed origin and methods, we can use config file when issue [#10](https://github.com/natemcmaster/dotnet-serve/issues/10) is implemented.
Is that ok?

Comment (natemcmaster) [reply src/dotnet-serve/Startup.cs]: Sounds good to me!

Comment (natemcmaster) [reply src/dotnet-serve/Startup.cs]: Oh, and thanks for the pull request 😄 I appreciate your willingness to contribute. Other than this one usage question, it looks great. You've done a good job!

Comment (ansariabr) [reply src/dotnet-serve/Startup.cs]: Welcome, and thanks 😉. I will update the code based on the suggestion.

**Comment by ansariabr** [comment]
Based on the logs in CI, I think the tests are failing on Mac and Linux because, the CI server is unable to start the dotnet-serve on the specified port. Can you please verify if it is because of that or something else @natemcmaster ?

**Comment by natemcmaster** [comment]
I don’t see what’s going wrong in tests based on inspection alone, but it appears consistent, even after retries. Will need to test more when I have my laptop.

File: src/dotnet-serve/Startup.cs [hunk 2]
```diff
@@ -62,6 +67,15 @@ public void ConfigureServices(IServiceCollection services)
 
         public void Configure(IApplicationBuilder app)
         {
+            if (_options.EnableCors)
+            {
+                app.UseCors(corsPolicy =>
```
Comment (natemcmaster) [inline src/dotnet-serve/Startup.cs]: Would it help us debug failures if this came after statuscodepages and developerexception middleware?

Comment (ansariabr) [reply src/dotnet-serve/Startup.cs]: Tried that, still failing. I created Github Action on my fork, this is what I am getting in exception of the test case which I have written to test OPTIONS request.
image

The test case passed on windows image in my Github Action too. It is failing due to newly added `-c` option, but again it is failing on Linux and Mac and working on Windows 😢.

Comment (natemcmaster) [reply src/dotnet-serve/Startup.cs]: The problem is that your test code isn't retrying long enough. It takes a second or two for dotnet-serve to start accepting requests. I use retries in all other tests. See https://github.com/natemcmaster/dotnet-serve/blob/master/test/dotnet-serve.Tests/HttpClientExtensions.cs

Comment (ansariabr) [reply src/dotnet-serve/Startup.cs]: Implemented the retrying mechanism, and now all the build is passing.

Comment (natemcmaster) [reply src/dotnet-serve/Startup.cs]: Thanks for fixing! I'll get this released to nuget.org momentarily...

**Pull request merged**